### PR TITLE
HW

### PR DESCRIPTION
**Pull request by stukalovsaleksandrs**

### CODE REVIEW

File: main_test.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	"testing"
+)
+
+type testCase struct {
+	name    string
+	getTree func() (root *node)
+	want    []int
+}
+
+var testCases = []testCase{
```
Comment (zayac) [inline main_test.go]: I'm expecting more tests. There are more corner cases that you test.
